content/md/en/docs/main-docs/build/build-process.md

```diff
@@ -1,118 +1,155 @@
-However, when starting a chain in developer mode at block 0, it uses the embedded Wasm from the native runtime.
+The WebAssembly execution environment can be more restrictive than the Rust execution environment.
+For example, the WebAssembly execution environment is a 32-bit architecture with a maximum 4GB of memory.
+Logic that can be executed in the WebAssembly runtime can always be executed in the Rust execution environment.
+However, not all logic that can be executed in the Rust runtime can be executed in the WebAssembly runtime.
+Block authoring nodes typically use the WebAssembly execution environment to help ensure that they produce valid blocks.
 
-There are several ways to configure a chain to meet your specific requirements:
+### Rust execution environment
 
-- `SKIP_WASM_BUILD` - Skips building any Wasm binary. This is useful when only native should be recompiled.
-    If this is the first run and there doesn't exist a Wasm binary, this will set both variables to `None`.
+As noted in [Default execution strategies](#default-execution-strategies), the Rust execution environment is used in some parts of the block handling process.
+However, this is only true if the Rust execution environment is compatible with the WebAssembly [runtime version](/main-docs/build/upgrade/#runtime-versioning).
+
+For most execution processes—except block authoring—the Rust execution environment is preferred because it provides better performance and has fewer restriction.
+However, you can override the default execution strategy for ny part of the block handling process by specifying command-line options.
+For example, if you want to use the Rust execution environment for block authoring, you can specify add `--execution-block-construction Native` to the command you use to start a node.
+
+## Building WebAssembly without a Rust runtime
+
+A WebAssembly runtime is required to start a new chain.
+After an initial WebAssembly runtime is provided, the blob that represents the WebAssembly runtime can be passed to other nodes as part of a [chain specification](/main-docs/build/chain-spec).
+In some rare cases, you might want to compile the WebAssembly target without the Rust runtime.
+For example, if you're testing a WebAssembly runtime to prepare for a forkless upgrade, you might want to compile just the new WebAssembly binary.
+
+Although it's a rare use case, you can use the [build-only-wasm.sh](https://github.com/paritytech/substrate/blob/master/.maintain/build-only-wasm.sh) script to build the `no_std` WebAssembly binary without compiling the Rust runtime.
+
+You can also use the `wasm-runtime-overrides` command-line option to load the WebAssembly from the filesystem.
+
+Usually when performing a runtime upgrade, you want to provide both a native and Wasm binary.
 
-- Use [this script](https://github.com/paritytech/substrate/blob/master/.maintain/build-only-wasm.sh) to build the `no_std` Wasm binary only. 
+## Compiling Rust without a WebAssembly runtime
```

Comment:
This is not really accurate. This is how you compile the rust executable, skipping the wasm compiling step.

You will still be "inserting" a wasm into your code, which is required for the native client, but it will be the previously compiled wasm, rather than a new one. 

This is just to increase the speed of quick development, where you do not need the wasm.